src/components/cylc/Graph.vue

```diff
@@ -0,0 +1,1644 @@
+      }
+    },
+
+    workflows: {
+      handler: function (newval, oldval) {
+        initialised ? this.workflowUpdated(newval) : console.debug('initialising')
+        initialised = true
+      },
+      deep: true
+    }
+  },
+
+  mixins: [mixin],
+
+  metaInfo () {
+    return {
+      title: this.getPageTitle('App.graph', { name: this.workflowId })
+    }
+  },
+
+  beforeRouteLeave (to, from, next) {
+    if (tippy) {
+      tippy.hide()
+    }
+    next()
+  },
+
+  beforeDestroy () {
+    if (tippy) {
+      tippy.hide()
+    }
+    cytoscapeService.unregister(this)
+  },
+
+  mounted () {
+    console.debug(`MOUNTED called, status: ${this.status}`)
+    this.handleMounted()
+    this.$store.watch((store) => {
+      this.workflows = store.workflows
+    })
+  },
+
+  created (cy) {
+    console.debug('CREATED')
+    this.suiteName = this.$route.params.workflowid
+    this.workflowId = this.$route.params.workflowid
+    cytoscapeService.register(
+      this,
+      {
+        activeCallback: this.setActive
+      }
+    )
+    this.subscribe('root')
+  },
+
+  components: {
+    SyncLoader,
+    cytoscape: VueCytoscape
+  },
+
+  methods: {
+    subscribe (queryName) {
+      const id = cytoscapeService.subscribe(
+        this,
+        QUERIES[queryName].replace('WORKFLOW_ID', this.workflowId)
+      )
+      if (!(queryName in this.subscriptions)) {
+        this.subscriptions[queryName] = {
+          id
+        }
+      }
+    },
+
+    unsubscribe (queryName) {
+      if (queryName in this.subscriptions) {
+        cytoscapeService.unsubscribe(
+          this.subscriptions[queryName].id
+        )
+      }
+    },
+
+    async workflowUpdated (workflows) {
+      try {
+        if (this.freeze === false) {
+          this.loading = true
+          const elements = {
+            nodes: [],
+            edges: []
+          }
+
+          if (!isUndefined(workflows)) {
+            each(workflows, (value, key) => {
+              each(value, (workflow, key) => {
+                if (!isEmpty(workflow.owner)) {
+                  this.workflowOwner = workflow.owner
+                }
+                if (!isEmpty(workflow.name)) {
+                  this.workflowName = workflow.name
+                }
+                if (has(workflow.nodesEdges, 'edges') && !isUndefined(workflow.nodesEdges.edges)) {
+                  elements.edges = this.getEdges(workflow.nodesEdges.edges)
+                }
+                if (has(workflow.nodesEdges, 'nodes') && !isUndefined(workflow.nodesEdges.nodes)) {
+                  elements.nodes = this.getNodes(workflow.nodesEdges.nodes)
+                }
+              })
+            })
+          }
+          console.debug('elements ==>>>> ', elements)
+          isEmpty(elements) ? console.warn('gdata is empty or undefined') : this.graphData = elements
+        }
+      } catch (error) {
+        console.error('workflowUpdated error: ', error)
+      }
+    },
```

Comment:
IMO we could move this method elsewhere, and instead just receive the data that the component needs via props. Even the parent view would be a better place, or copy what's done in the tree view and add a hook in the vuex store too.

The Vuex store sets a global `.workflows` value. After the PR for the family tree, it will construct the `workflowTree` iff the `workflows.familyProxies` exist. That attribute - so far - is only ever present when you have a Tree view.

Similarly, we could call the method to create the Graph data iff the `workflows.nodesEdges` exist. This way, if you are on a view with only Trees, the graph method is not called, and vice-versa. And, if you are on the Dashboard, probably none will be ever called, as it will probably need just the `workflows`.

Reply:
If we have flat data that doesn't need any manipulation then this method isn't needed, it can be sent straight to updateGraph().  I was using a watch on the store as using the 'deep' option can watch for changes deep in the data hierarchy.  Hopefully in the future  it will be just change data, so any change at all would be acted upon.